<ui version="4.0" >
 <class>toBrowserConstraintUI</class>
 <widget class="QDialog" name="toBrowserConstraintUI" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>669</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Constraint info</string>
  </property>
  <layout class="QGridLayout" >
   <property name="leftMargin" >
    <number>11</number>
   </property>
   <property name="topMargin" >
    <number>11</number>
   </property>
   <property name="rightMargin" >
    <number>11</number>
   </property>
   <property name="bottomMargin" >
    <number>11</number>
   </property>
   <property name="horizontalSpacing" >
    <number>6</number>
   </property>
   <property name="verticalSpacing" >
    <number>6</number>
   </property>
   <item row="2" column="0" colspan="4" >
    <layout class="QHBoxLayout" >
     <item>
      <widget class="QComboBox" name="Name" >
       <property name="sizePolicy" >
        <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
         <horstretch>3</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="Type" >
       <property name="sizePolicy" >
        <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <item>
        <property name="text" >
         <string>Foreign Key</string>
        </property>
       </item>
       <item>
        <property name="text" >
         <string>Check</string>
        </property>
       </item>
       <item>
        <property name="text" >
         <string>Unique Primary Key</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item row="4" column="1" >
    <spacer>
     <property name="orientation" >
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeType" >
      <enum>QSizePolicy::Expanding</enum>
     </property>
     <property name="sizeHint" >
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="0" colspan="4" >
    <layout class="QHBoxLayout" >
     <item>
      <widget class="QLabel" name="TextLabel1" >
       <property name="sizePolicy" >
        <sizepolicy vsizetype="Preferred" hsizetype="Expanding" >
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text" >
        <string>&amp;Name</string>
       </property>
       <property name="wordWrap" >
        <bool>false</bool>
       </property>
       <property name="buddy" >
        <cstring>Name</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="AddButton" >
       <property name="text" >
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="DeleteButton" >
       <property name="text" >
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="4" column="3" >
    <widget class="QPushButton" name="PushButton2" >
     <property name="text" >
      <string>Cancel</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0" colspan="4" >
    <widget class="QStackedWidget" name="WidgetStack" >
     <property name="sizePolicy" >
      <sizepolicy vsizetype="Expanding" hsizetype="Preferred" >
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <widget class="QWidget" name="ReferentialPage" >
      <layout class="QGridLayout" >
       <item rowspan="2" row="1" column="0" >
        <widget class="QLabel" name="TextLabel3" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text" >
          <string>&amp;Columns</string>
         </property>
         <property name="wordWrap" >
          <bool>false</bool>
         </property>
         <property name="buddy" >
          <cstring>SourceColumns</cstring>
         </property>
        </widget>
       </item>
       <item row="3" column="0" >
        <widget class="QLineEdit" name="SourceColumns" />
       </item>
       <item row="4" column="0" >
        <widget class="toResultCols" native="1" name="SourceColList" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Expanding" hsizetype="Fixed" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="focusPolicy" >
          <enum>Qt::StrongFocus</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="2" colspan="2" >
        <widget class="QLabel" name="TextLabel5" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text" >
          <string>R&amp;efered columns</string>
         </property>
         <property name="wordWrap" >
          <bool>false</bool>
         </property>
         <property name="buddy" >
          <cstring>ReferedColumns</cstring>
         </property>
        </widget>
       </item>
       <item row="3" column="2" colspan="2" >
        <widget class="QLineEdit" name="ReferedColumns" />
       </item>
       <item row="4" column="2" colspan="2" >
        <widget class="toResultCols" native="1" name="ReferColList" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Expanding" hsizetype="Fixed" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="focusPolicy" >
          <enum>Qt::StrongFocus</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="2" colspan="2" >
        <widget class="Line" name="Line3" >
         <property name="frameShape" >
          <enum>QFrame::HLine</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item rowspan="5" row="0" column="1" >
        <widget class="Line" name="Line1" >
         <property name="frameShape" >
          <enum>QFrame::VLine</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="3" >
        <widget class="QComboBox" name="ReferTable" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="editable" >
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="2" >
        <widget class="QLabel" name="TextLabel4" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Minimum" hsizetype="Preferred" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text" >
          <string>Refers &amp;To</string>
         </property>
         <property name="wordWrap" >
          <bool>false</bool>
         </property>
         <property name="buddy" >
          <cstring>ReferTable</cstring>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="PrimaryPage" >
      <layout class="QVBoxLayout" >
       <item>
        <widget class="QGroupBox" name="UniqueType" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Maximum" hsizetype="Preferred" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title" >
          <string>&amp;Type</string>
         </property>
         <layout class="QGridLayout" >
          <property name="leftMargin" >
           <number>11</number>
          </property>
          <property name="topMargin" >
           <number>11</number>
          </property>
          <property name="rightMargin" >
           <number>11</number>
          </property>
          <property name="bottomMargin" >
           <number>11</number>
          </property>
          <property name="horizontalSpacing" >
           <number>6</number>
          </property>
          <property name="verticalSpacing" >
           <number>6</number>
          </property>
          <item row="0" column="0" >
           <widget class="QRadioButton" name="Primary" >
            <property name="text" >
             <string>Primary key</string>
            </property>
            <property name="checked" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="1" >
           <widget class="QRadioButton" name="Unique" >
            <property name="text" >
             <string>Unique</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="TextLabel2" >
         <property name="text" >
          <string>&amp;Columns</string>
         </property>
         <property name="wordWrap" >
          <bool>false</bool>
         </property>
         <property name="buddy" >
          <cstring>UniqueCols</cstring>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="UniqueCols" />
       </item>
       <item>
        <widget class="toResultCols" native="1" name="UniqueColList" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Expanding" hsizetype="Preferred" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="focusPolicy" >
          <enum>Qt::StrongFocus</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="CheckPage" >
      <layout class="QGridLayout" >
       <item row="2" column="0" >
        <widget class="QLabel" name="TextLabel2_2" >
         <property name="text" >
          <string>C&amp;ondition</string>
         </property>
         <property name="wordWrap" >
          <bool>false</bool>
         </property>
         <property name="buddy" >
          <cstring>CheckCondition</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="0" >
        <widget class="QLabel" name="TextLabel6" >
         <property name="text" >
          <string>&amp;Column</string>
         </property>
         <property name="wordWrap" >
          <bool>false</bool>
         </property>
         <property name="buddy" >
          <cstring>CheckColumn</cstring>
         </property>
        </widget>
       </item>
       <item row="3" column="0" >
        <widget class="QLineEdit" name="CheckCondition" />
       </item>
       <item row="1" column="0" >
        <widget class="toResultCols" native="1" name="CheckColumn" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Expanding" hsizetype="Preferred" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="focusPolicy" >
          <enum>Qt::StrongFocus</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="4" column="2" >
    <widget class="QPushButton" name="ApplyButton" >
     <property name="text" >
      <string>&amp;Ok</string>
     </property>
     <property name="default" >
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="4" column="0" >
    <widget class="QToolButton" name="ToolButton1" >
     <property name="focusPolicy" >
      <enum>Qt::StrongFocus</enum>
     </property>
     <property name="toolTip" >
      <string>Display SQL to create this constraint</string>
     </property>
     <property name="text" >
      <string/>
     </property>
    </widget>
   </item>
   <item row="0" column="0" colspan="4" >
    <widget class="toTableSelect" native="1" name="TableSelect" />
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11" />
 <customwidgets>
  <customwidget>
   <class>toResultCols</class>
   <extends>QWidget</extends>
   <header>toresultcols.h</header>
  </customwidget>
  <customwidget>
   <class>toTableSelect</class>
   <extends>QWidget</extends>
   <header>totableselect.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>Name</tabstop>
  <tabstop>ReferTable</tabstop>
  <tabstop>SourceColumns</tabstop>
  <tabstop>ReferedColumns</tabstop>
  <tabstop>SourceColList</tabstop>
  <tabstop>ReferColList</tabstop>
  <tabstop>Primary</tabstop>
  <tabstop>UniqueCols</tabstop>
  <tabstop>UniqueColList</tabstop>
  <tabstop>CheckColumn</tabstop>
  <tabstop>CheckCondition</tabstop>
  <tabstop>ToolButton1</tabstop>
  <tabstop>ApplyButton</tabstop>
  <tabstop>PushButton2</tabstop>
 </tabstops>
 <includes>
  <include location="local" >toresultcols.h</include>
  <include location="local" >toresultcols.h</include>
  <include location="local" >toresultcols.h</include>
  <include location="local" >toresultcols.h</include>
  <include location="local" >totableselect.h</include>
 </includes>
 <resources/>
 <connections>
  <connection>
   <sender>ToolButton1</sender>
   <signal>clicked()</signal>
   <receiver>toBrowserConstraintUI</receiver>
   <slot>displaySQL()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PushButton2</sender>
   <signal>clicked()</signal>
   <receiver>toBrowserConstraintUI</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ApplyButton</sender>
   <signal>clicked()</signal>
   <receiver>toBrowserConstraintUI</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Name</sender>
   <signal>activated(int)</signal>
   <receiver>toBrowserConstraintUI</receiver>
   <slot>changeConstraint()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AddButton</sender>
   <signal>clicked()</signal>
   <receiver>toBrowserConstraintUI</receiver>
   <slot>addConstraint()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DeleteButton</sender>
   <signal>clicked()</signal>
   <receiver>toBrowserConstraintUI</receiver>
   <slot>delConstraint()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ReferTable</sender>
   <signal>activated(QString)</signal>
   <receiver>toBrowserConstraintUI</receiver>
   <slot>changeRefered()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Type</sender>
   <signal>activated(int)</signal>
   <receiver>toBrowserConstraintUI</receiver>
   <slot>changeType(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
